cosmwasm/contracts/crosschain-registry/src/execute.rs

```diff
@@ -0,0 +1,851 @@
+use crate::helpers::*;
+use crate::state::{CHAIN_TO_CHAIN_CHANNEL_MAP, CHANNEL_ON_CHAIN_CHAIN_MAP, CONTRACT_ALIAS_MAP};
+use cosmwasm_std::{Deps, DepsMut, Response, StdError};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::ContractError;
+
+// Enum to represent the operation to be performed
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+pub enum Operation {
+    Set,
+    Change,
+    Remove,
+}
+
+// Contract Registry
+
+// Struct for input data for a single contract alias
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+pub struct ContractAliasInput {
+    pub operation: Operation,
+    pub alias: String,
+    pub address: Option<String>,
+    pub new_alias: Option<String>,
+}
+
+// Set, change, or remove a contract alias
+pub fn contract_alias_operations(
+    deps: DepsMut,
+    operations: Vec<ContractAliasInput>,
+) -> Result<Response, ContractError> {
+    let response = Response::new();
+    for operation in operations {
+        match operation.operation {
+            Operation::Set => {
+                if CONTRACT_ALIAS_MAP.has(deps.storage, &operation.alias) {
```

Comment:
we will probably want separate functions for each of these when we add gov handling

Reply:
Discussed offline with Nicolas, if we decide to do this it can be in a separate PR. Please unresolve if this changes.